bpf/lib/utime.h

```diff
@@ -0,0 +1,41 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/* Copyright Authors of Cilium */
+
+#ifndef __LIB_UTIME_H_
+#define __LIB_UTIME_H_
+
+#include "common.h"
+#include "maps.h"
+
+/*
+ * Number of bits to shift a monotonic 64-bit nanosecond clock for utime unit.
+ * Dividing nanoseconds by 2^9 yields roughly half microsecond accuracy without
+ * any loss when converting from seconds (as 1e9/2^9 is an integer), but avoids
+ * expensive 64-bit divisions.  With this shift the range of an u64 is ~300000
+ * years instead of ~600 years if left at milliseconds.
```

Comment:
```suggestion
 * Number of bits to shift a monotonic 64-bit nanosecond clock for utime unit.
 * Dividing nanoseconds by 2^9 yields roughly half microsecond accuracy without
 * any loss when converting from seconds (as 1e9/2^9 is an integer), but avoids
 * expensive 64-bit divisions.  With this shift the range of an u64 is ~300000
 * years instead of ~600 years if left at nanoseconds.
```
?

Reply:
Incidentally, while the last statement is technically true, we wouldn't be able to initialize or make `utime_get_time()` calculations based on a value that represents >~600 years currently because we first start with a timestamp in nanoseconds, thereby restricting us to `EPOCH < ktime_ns < ~600y`, then apply the math to render this into UTIME.

Not really sure there's much actionable out of this observation, other than to say that the folks dealing with the "Year 2554" problem should have fun with it.

Reply:
Fixed the comment, thanks!

While we start with a timestamp in nanoseconds, that is nanoseconds since boot, so if the system is rebooted every 600 years, utime can and will represent Unix epoch based expirations times upto year 301460 :-) Also the userspace tooling translating a given expiration `time.TIme` to `utime` units never changes the timestamp to nanoseconds only. There is also a unit test that checks that the accuracy of timestamp translations from `time.Time` to `utime` and back to `time.Time` keeps microsecond accuracy from 1970 (second 0) to year 301460 (second 944_4732_965_739) iterating a large prime and using the value both as seconds and nanoseconds to not test only whole secnods.

Reply:
💡 Ah right, I don't know why I assumed it was time since boot elsewhere then switched my assumption to time since Unix Epoch here. Thanks for explaining!